### ISSUE TITLE
"Duplicate coordinates found" error when initializing "SparseTensor"

### ISSUE THREAD

**Issue by jingyi-zhang** [NVIDIA/MinkowskiEngine#116]
Thanks for your good code！
but I got an error  occasionally when use function 'SparseTensor'
I dont know what caused this error
ca
![image](https://user-images.githubusercontent.com/52694560/77853651-c699bd80-7217-11ea-906f-186d287de4d6.png)

**Comment by jingyi-zhang** [comment]
This is a part code in '__getitem__' function of DataSet
`		coords = lidar_camera[:,0:2]
		#print(coords.shape[0])
		features = lidar_camera[:,2:4]
		labels = object_list
		coords /= self.quantization_size
		#print(coords.shape)
		mapping = ME.utils.sparse_quantize(coords=coords,
			return_index=True)
		#print(len(mapping))
		return coords[mapping], features[mapping], labels`

this my test code
`
train_set = SparseKittiDataset(classes=FLAGS.classes, split='training', augment=False, cfg=cfg)
train_loader = DataLoader(dataset=train_set, batch_size=FLAGS.batch_size, num_workers=FLAGS.workers,\
	shuffle=True, drop_last=True,collate_fn=ME.utils.batch_sparse_collate)
for batch_idx, data_label in enumerate(train_loader):
	coords, features, label = data_label
	#print(features.shape)
	#print(coords.shape)
	input = ME.SparseTensor(features, coords, force_creation=True)
	print(input)
	break`

**Comment by chrischoy** [comment]
It happens when you have duplicate coordinates.

The standard way to fix this error would be sampling one point per voxel. This process is known as the voxel downsampling.

The engine supports such downsampling and quantization. Please refer to [sparse_quantize](https://stanfordvl.github.io/MinkowskiEngine/utils.html#MinkowskiEngine.utils.sparse_quantize)

In most cases, a training pipeline consists of

1. quantization https://github.com/StanfordVL/MinkowskiEngine/blob/415098d4e457a1373e0a0499414cee0cbfbd6144/examples/training.py#L139
2. collation and batched coordinate generation https://github.com/StanfordVL/MinkowskiEngine/blob/415098d4e457a1373e0a0499414cee0cbfbd6144/examples/training.py#L154
3. SparseTensor generation https://github.com/StanfordVL/MinkowskiEngine/blob/415098d4e457a1373e0a0499414cee0cbfbd6144/examples/training.py#L199

Please let me know if you have further questions.

**Comment by chrischoy** [comment]
In your example, you missed `batched_coordinates` which was in the collation function.

This results in 2D sparse tensor. Coordinates must have size N x 4, not N x 3 to create a 3D sparse tensor.

**Comment by jingyi-zhang** [comment]
Thanks for your patience!
You are so nice~

Took your advice, I used 'batched_coordinates' instead of 'batch_sparse_collate' to load data, then, the error has been solved.

But actually, I am still confused about the difference between 'batch_sparse_collate', 'SparseCollation' and 'batched_coordinates', I checked their definition in 'collation.py', seems like they have same function.
